### ISSUE TITLE
[0.9.3 built off master] Tons of shard groups being created

### ISSUE THREAD

**Issue by mjdesa** [influxdb/influxdb#3594]
After reproducing https://github.com/influxdb/influxdb/issues/3572 I noticed a ton of new shard groups being created for the databases I made for reproducing the issue.

```
[retention] 2015/08/07 16:54:00 retention policy enforcement check commencing
[retention] 2015/08/07 16:54:00 retention policy shard deletion check commencing
[metastore] 2015/08/07 17:00:58 new shard group 906 successfully created for database bigsmall, retention policy default
[metastore] 2015/08/07 17:00:58 new shard group 907 successfully created for database testcase, retention policy default
[retention] 2015/08/07 17:04:00 retention policy enforcement check commencing
[retention] 2015/08/07 17:04:00 retention policy shard deletion check commencing
[metastore] 2015/08/07 17:10:58 new shard group 908 successfully created for database bigsmall, retention policy default
[metastore] 2015/08/07 17:10:58 new shard group 909 successfully created for database testcase, retention policy default
```

**Comment by beckettsean** [comment]
What was the retention policy for the db, how long had it been running, and how many points written?

**Comment by mjdesa** [comment]
default INF

The same point was written to about 5 times. No other points were written.

**Comment by otoolep** [comment]
Can you show a sequence of `curl` commands that show this happening?

There are two ways shards may be created -- because a) point's timestamp is destined for a slice of time for which there are no shards, and b) shard precreation, which attempts to create shards before they are needed:

https://github.com/influxdb/influxdb/blob/master/services/precreator/README.md

**Comment by mjdesa** [comment]
1. `curl -G "http://localhost:8086/query" --data-urlencode "q=CREATE DATABASE example"`

2. `curl -i -XPOST "http://localhost:8086/write?db=example" --data-binary 'updown value=100 1434055562000000000'`

3. `curl -i -XPOST "http://localhost:8086/write?db=example" --data-binary 'updown value=50 1434055562000000000'`

4. `curl -i -XPOST "http://localhost:8086/write?db=example" --data-binary 'updown value=200 1434055562000000000'`

5. `curl -i -XPOST "http://localhost:8086/write?db=example" --data-binary 'updown value=55 1434055562000000000'`

6. `curl -G "http://localhost:8086/query" --data-urlencode "db=example" --data-urlencode "q=SELECT * FROM updown"`

7. Kill inlfuxd

8. `curl -G "http://localhost:8086/query" --data-urlencode "db=example" --data-urlencode "q=SELECT * FROM updown"`

9. `curl -i -XPOST "http://localhost:8086/write?db=example" --data-binary 'updown value=100 1434055562000000000'`

10. Then every 10 minutes new shard groups get created.

I'll comment in 10 minutes if I start getting logs.

**Comment by mjdesa** [comment]
Shard groups for the three databases that I wrote smaller and larger values with the same timestamp,

```
[metastore] 2015/08/10 10:14:15 new shard group 943 successfully created for database bigsmall, retention policy default
[metastore] 2015/08/10 10:14:15 new shard group 944 successfully created for database testcase, retention policy default
[metastore] 2015/08/10 10:14:15 new shard group 945 successfully created for database example, retention policy default
```

**Comment by jwilder** [comment]
Not sure there is an issue here.  Looks like shard recreation service which defaults to running every 10m.  Shard group IDs are global across all DBs too so  `945` just means that the counter is at `945` now.

**Comment by beckettsean** [comment]
@mjdesa maybe this is a duplicate of https://github.com/influxdb/influxdb/issues/3722?

**Comment by otoolep** [comment]
Yes, if you write data into the past shards will be created to fill in the entire time from the point in the past to "now" (whatever now is).

**Comment by otoolep** [comment]
Shard precreation can be disabled if necessary, if it's just a hassle for testing.

**Comment by mjdesa** [comment]
@beckettsean I think it is.

**Comment by beckettsean** [comment]
closed in favor of #3722
